tests/acceptance/welcome/payment-info-test.js

```diff
@@ -0,0 +1,84 @@
+import Ember from 'ember';
+import startApp from '../../helpers/start-app';
+import { stripeMock } from '../../../utils/stripe';
+import { stubRequest } from "../../helpers/fake-server";
+
+var application;
+
+module('Acceptance: WelcomePaymentInfo', {
+  setup: function() {
+    application = startApp();
+  },
+  teardown: function() {
+    stripeMock.createStripeToken = null;
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('visiting /welcome/payment-info when not logged in', function() {
+  visit('/welcome/payment-info');
+
+  andThen(function() {
+    equal(currentPath(), 'login');
+  });
+});
+
+test('submitting empty payment info raises an error', function() {
+  stripeMock.createStripeToken = function() {
+    return Ember.RSVP.reject({ message: 'Failure' });
+  };
+
+  stubOrganizations();
+
+  signInAndVisit('/welcome/payment-info');
+  click('button:contains(Save)');
+
+  andThen(function() {
+    equal(currentPath(), 'welcome.payment-info');
+    var error = find('p:contains(Failure)');
+    ok(error.length, 'errors are on the page');
+  });
+});
+
+test('submitting valid payment info should be successful', function() {
+  // This is to load apps.index
+  stubStacks();
+  var name = 'Bob Boberson';
+  var cardNumber = '4242424242424242';
+  var cvc = '123';
+  var expMonth = '03';
+  var expYear = '2019';
+  var addressZip = '11111';
+  var stripeToken = 'some-token';
+
+  stubRequest('post', '/organizations/1/subscriptions', function(request){
+    var params = JSON.parse(request.requestBody);
+    equal(params.stripe_token, stripeToken, 'stripe token is correct');
+    return this.success();
+  });
+
+  stubOrganizations();
+
+  stripeMock.createStripeToken = function(options) {
+    equal(options.name, name, 'name is correct');
+    equal(options.number, cardNumber, 'card number is correct');
+    equal(options.cvc, cvc, 'cvc is correct');
+    equal(options.exp_month, expMonth, 'exp month is correct');
+    equal(options.exp_year, expYear, 'exp year is correct');
+    equal(options.address_zip, addressZip, 'zip is correct');
+    return Ember.RSVP.resolve({ id: stripeToken });
+  };
+
+  signInAndVisit('/welcome/payment-info');
+  fillIn('[name=name]', name);
+  fillIn('[name=number]', cardNumber);
+  fillIn('[name=cvc]', cvc);
+  fillIn('[name=exp-month]', expMonth);
+  fillIn('[name=exp-year]', expYear);
+  fillIn('[name=zip]', addressZip);
+  click('button:contains(Save)');
+
+  andThen(function() {
+    equal(currentPath(), 'apps.index');
+  });
```

Comment:
this test would probably benefit from an `expect(8)` at the beginning — it's not clear that `stripeMock.createStripeToken` will necessarily be called, and the test could still pass if it were never called. I.e., someone could change the internals of `app/utils/stripe.js` to avoid calling `stripeMock.createStripeToken`, but this test wouldn't fail.